pkg/diffs/diffs.go

```diff
@@ -51,13 +57,36 @@ func GetChangedCiopConfigs(masterConfig, prConfig config.CompoundCiopConfig, log
 			continue
 		}
 
-		if !equality.Semantic.DeepEqual(oldConfig, newConfig) {
+		withoutTestsOldConfig := cioperatorapi.ReleaseBuildConfiguration{}
```

Comment:
There's a lot going on here -- can't we loop over the tests, and if none are found to be different, then use a top-level check?

```go
		if !equality.Semantic.DeepEqual(oldConfig, newConfig) {
			ret[filename] = newConfig
		}
		affectedJobs[filename] = sets.NewString()
		oldTests := getTestsByName(oldConfig.Tests)
		for as, test := range getTestsByName(newConfig.Tests) {
			if !equality.Semantic.DeepEqual(oldTests[as], test) {
				logger.WithField(logCiopConfig, filename).Info(changedCiopConfigMsg)
				affectedJobs[filename].Insert(as)
			}
		}
```

Reply:
There is always a chance that the tests are changed but other fields as well. In that case, we want to rehearse all jobs.

Reply:
True!